fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/directory/FDBIndexOutput.java

```diff
@@ -100,8 +98,8 @@ public FDBIndexOutput(@Nonnull String resourceDescription, @Nonnull String name,
         this.fdbDirectory = fdbDirectory;
         blockSize = fdbDirectory.getBlockSize();
         buffer = BUFFERS.poll();
-        if (buffer == null) {
-            buffer = ByteBuffer.allocate((int)blockSize);
+        if (buffer == null || buffer.capacity() != blockSize) {
+            buffer = ByteBuffer.allocate(blockSize);
```

Comment:
It may be interesting to know how many of these allocations we've done

Reply:
It is my understanding that adding log messages here will be a flood. I wouldn't like to do it in the same PR. 
(Note that this buffer can also be created after the first pool initialization)